pkg/cli/mirror/copy.go

```diff
@@ -364,45 +370,165 @@ func getConfigPathFromConfigLayer(imagePath, configSha string) (string, error) {
 	}
 	return "", fmt.Errorf("label %s not found in config blob %s", configsLabel, configLayerDir)
 }
+func getRelatedImagesFromBundle(bundle declcfg.Bundle, packages []v1alpha2.IncludePackage) ([]declcfg.RelatedImage, error) {
+	images := []declcfg.RelatedImage{}
+	uniqueImages := []declcfg.RelatedImage{}
 
-// getRelatedImages this reads each catalog or config.json
-// file in a given operator in the FBC
-// and returns the list of relatedImages found in the CSV
-func getRelatedImages(path string) ([]model.RelatedImage, error) {
-	var icJSON *model.Bundle
-
-	// read the config.json to get releated images
-	icd, err := ioutil.ReadFile(path + catalogJSON)
-	if err != nil {
-		return nil, err
-	}
-	// we are only interested in the related images section
-	tmp := string(icd)
-	i := strings.Index(tmp, relatedImages)
-	newJson := "{\"" + tmp[i:]
-	err = json.Unmarshal([]byte(newJson), &icJSON)
-	if err != nil {
-		return nil, fmt.Errorf("json unmarshal %v", err)
-	}
-	//unique images - remove duplicate values
-	images := []model.RelatedImage{}
-
-	for _, v := range icJSON.RelatedImages {
+	for _, v := range bundle.RelatedImages {
 		found := false
-		for _, w := range images {
+		for _, w := range uniqueImages {
 			if w.Image == v.Image {
 				found = true
 				break
 			}
 		}
 		if !found {
-			images = append(images, v)
+			uniqueImages = append(uniqueImages, v)
 		}
 	}
 
+	if len(packages) > 0 {
+		for _, pkg := range packages {
+			if bundle.Package == pkg.Name {
+				images = append(images, uniqueImages...)
+			}
+		}
+	} else {
+		images = append(images, uniqueImages...)
+	}
 	return images, nil
 }
 
+// getRelatedImages this reads each catalog or config.json
+// file in a given operator in the FBC
+// and returns the list of relatedImages found in the CSV
+func getRelatedImages(root string, packages []v1alpha2.IncludePackage) ([]declcfg.RelatedImage, error) {
```

Comment:
It may be easier to do this logic using the opm libraries with `declcfg.LoadFS` or `declcfg.WalkFS` https://github.com/operator-framework/operator-registry/blob/a0bb1e5b6d1ad34154fe205401dbd6ee6c4276bb/alpha/declcfg/load.go#L30. Converting to the model would probably be easier to work with the declfg since it stored the object in maps instead of slices https://github.com/operator-framework/operator-registry/blob/a0bb1e5b6d1ad34154fe205401dbd6ee6c4276bb/alpha/declcfg/declcfg_to_model.go#L13.

Reply:
You have a great point @jpower432 . If we can rely on those functions, we can reduce the complexity a lot.
The first time we tried the `declcfg.LoadFS` and `WalkFS` (https://github.com/openshift/oc-mirror/pull/511), we didn't quite make it. 
Since we only have 1 more sprint to go, and we're supposed to focus on fixing bugs, do you think we could ship the TechPreview version as is, and plan to replace with  `declcfg.LoadFS` and `WalkFS` with 4.13?
WDYT?

Reply:
@sherine-k Seems fine from my perspective since you have unit tests that will enable this to be refactored.